postgres_exporter/common/setup_metric_views.sql

```diff
@@ -0,0 +1,174 @@
+-- #########################
+-- Metric View Objects
+-- #########################
+
+CREATE SCHEMA IF NOT EXISTS monitor;
+
+-- Preserve existing data if users added more views or changed schedule. Allows setup call to be idempodent.
+DO $pgmonitor$
+DECLARE
+v_exists    smallint;
+BEGIN
+
+    SELECT count(*) INTO v_exists FROM information_schema.tables WHERE table_schema = 'monitor' AND table_name = 'metric_views';
+    IF v_exists > 0 THEN
+        CREATE TEMPORARY TABLE metric_views_preserve_temp (LIKE monitor.metric_views);
+        INSERT INTO metric_views_preserve_temp SELECT * FROM monitor.metric_views;
+        DROP TABLE monitor.metric_views;
+    END IF;
+
+    CREATE TABLE IF NOT EXISTS monitor.metric_views (
+        view_schema text NOT NULL DEFAULT 'monitor'
+        , view_name text NOT NULL
+        , concurrent_refresh boolean NOT NULL DEFAULT true
+        , run_interval interval NOT NULL
+        , last_run timestamptz
+        , active boolean NOT NULL DEFAULT true
+        , scope text NOT NULL default 'global'
+        , CONSTRAINT metric_views_pk PRIMARY KEY (view_schema, view_name)
+        , CONSTRAINT metric_views_scope_ck CHECK (scope IN ('global', 'database'))
+    );
+
+    IF v_exists > 0 THEN
+        INSERT INTO monitor.metric_views SELECT * FROM metric_views_preserve_temp;
+        DROP TABLE metric_views_preserve_temp;
+    END IF;
+
+END
+$pgmonitor$;
+
+
+DROP PROCEDURE IF EXISTS monitor.refresh_metric_views (text, text);
+CREATE PROCEDURE monitor.refresh_metric_views (p_view_schema text DEFAULT 'monitor', p_view_name text DEFAULT NULL)
```

Comment:
Looks like the other parts of this file are assuming that there's a hard-coded `monitor` schema (`CREATE SCHEMA`, literal `monitor` in the `SELECT` statement, etc).  Does it make sense for this to be customizable?

Reply:
All objects are currently hard coded to the monitor schema. Too challenging to make that optional at this time. If we ever get to making this into an actual extension for the db objects at some point, options for the schema name would be much easier.

Reply:
Sorry, I think I explained in a convoluted way; it appears that the procedure allows for customized schema, but the code itself does not use them. So is this anticipating future changes/enhancements then? It seems like if the only schema we care about is hard-coded at this point it would be unnecessary to have a parameter we pass in.